ihaskell-display/ihaskell-widgets/src/IHaskell/Display/Widgets/Button.hs

```diff
@@ -0,0 +1,210 @@
+{-# LANGUAGE FlexibleInstances #-}
+{-# LANGUAGE OverloadedStrings #-}
+
+module IHaskell.Display.Widgets.Button (
+    -- * The Button Widget
+    Button,
+    -- * Predefined button styles
+    ButtonStyle(..),
+    -- * Create a new button
+    mkButton,
+    -- * Set button properties
+    setButtonStyle,
+    setButtonLabel,
+    setButtonTooltip,
+    disableButton,
+    enableButton,
+    toggleButtonStatus,
+    -- * Get button properties
+    getButtonStyle,
+    getButtonLabel,
+    getButtonTooltip,
+    isDisabled,
+    -- * Click handlers
+    setClickHandler,
+    getClickHandler,
+    triggerClick,
+    ) where
+
+-- To keep `cabal repl` happy when running from the ihaskell repo
+import           Prelude
+
+import           Control.Monad (when)
+import           Data.Aeson (ToJSON, Value(..), object, toJSON, (.=))
+import           Data.Aeson.Types (Pair)
+import           Data.HashMap.Strict as Map
+import           Data.IORef
+import           Data.Text (Text)
+import qualified Data.Text as T
+import           System.IO.Unsafe (unsafePerformIO)
+
+import           IHaskell.Display
+import           IHaskell.Eval.Widgets
+import qualified IHaskell.IPython.Message.UUID as U
+import           IHaskell.Types (WidgetMethod(..))
+
+-- | ADT for a button
+data Button =
+       Button
+         { uuid :: U.UUID
+         , description :: IORef Text
```

Comment:
An explanation of where each of these texts and bools and so on are used would be nice. Very brief, but just like "text inside the button" or "shown on hover", etc.